devito/interfaces.py

```diff
@@ -255,3 +265,60 @@ def indices(cls, shape):
         """
         _indices = [t, x, y, z]
         return _indices[:len(shape) + 1]
+
+
+class MemmapManager():
+    """Class for managing all memmap related settins"""
+    # used to enable memmap as default
+    _use_memmap = False
+    # flag for registering exit func
+    _registered = False
+    _default_disk_path = "/tmp/devito_disk"
+    # contains str name of all memmap file created
+    _created_data = set()
+    _default_exit_code = 0
+
+    @staticmethod
+    def memmap_enabled_by_default():
+        """call this method to enable memmap by default"""
+        return MemmapManager._use_memmap
+
+    @staticmethod
+    def set_default_disk_path(default_disk_path):
+        """call this method to change the default disk path for memmap"""
+        MemmapManager._default_disk_path = default_disk_path
+
+    @staticmethod
+    def setup(data_self, *args, **kwargs):
+        """call this method with arguments self, args, kwargs to setup memmap parameters"""
+        disk_path = kwargs.get('disk_path', MemmapManager._default_disk_path)
+        if not os.path.exists(disk_path):
+            os.makedirs(disk_path)
+        data_self.f = disk_path + "/data_" + kwargs.get('name')
+        data_self.memmap = kwargs.get('memmap', False) or MemmapManager._use_memmap
+        MemmapManager._created_data.add(data_self.f)
+        if not MemmapManager._registered:
+            MemmapManager._register_remove_memmap_file_signal()
+            MemmapManager._registered = True
+
+    # clean up memmap file
+    @staticmethod
+    def _reomve_memmap_file():
+        for f in MemmapManager._created_data:
+            try:
+                os.remove(f)
+            except OSError:
+                print("error removing " + f + " skipping")
+                pass
+
+    # used for clean memmap file on signal, internal method
+    @staticmethod
+    def _remove_memmap_file_on_signal(*args):
```

Comment:
I am not sure I am reading this right. Where is it actually calling `_remove_memmap_file()` or removing the file in the `signal` invocation chain? Can you explain please how this is supposed to work?